relevanceai/operations/auto/cluster.py

```diff
@@ -0,0 +1,609 @@
+    #     preprocess_hooks: Optional[List[callable]] = None,
+    #     algorithm: str = "rake",
+    #     n: int = 2,
+    # ):
+    #     """
+    #     Simple implementation of the cluster word cloud.
+
+    #     Parameters
+    #     ------------
+    #     vector_fields: list
+    #         The list of vector fields
+    #     text_fields: list
+    #         The list of text fields
+    #     cluster_alias: str
+    #         The alias of the cluster
+    #     cluster_field: str
+    #         The cluster field to try things on
+    #     num_clusters: int
+    #         The number of clusters
+    #     preprocess_hooks: list
+    #         The preprocess hooks
+    #     algorithm: str
+    #         The algorithm to use
+    #     n: int
+    #         The number of words
+
+    #     """
+    #     preprocess_hooks = [] if preprocess_hooks is None else preprocess_hooks
+
+    #     vector_fields_str = ".".join(sorted(vector_fields))
+    #     field = f"{cluster_field}.{vector_fields_str}.{cluster_alias}"
+    #     all_clusters = self.facets([field], page_size=num_clusters)
+    #     cluster_counters = {}
+    #     if "results" in all_clusters:
+    #         all_clusters = all_clusters["results"]
+    #     # TODO: Switch to multiprocessing
+    #     for c in tqdm(all_clusters[field]):
+    #         cluster_value = c[field]
+    #         top_words = self.keyphrases(
+    #             text_fields=text_fields,
+    #             n=n,
+    #             filters=[
+    #                 {
+    #                     "field": field,
+    #                     "filter_type": "contains",
+    #                     "condition": "==",
+    #                     "condition_value": cluster_value,
+    #                 }
+    #             ],
+    #             most_common=most_common,
+    #             preprocess_hooks=preprocess_hooks,
+    #             algorithm=algorithm,
+    #         )
+    #         cluster_counters[cluster_value] = top_words
+    #     return cluster_counters
+
+    # TODO: Add keyphrases to auto cluster
+    # def auto_cluster_keyphrases(
+    #     vector_fields: List[str],
+    #     text_fields: List[str],
+    #     cluster_alias: str,
+    #     deployable_id: str,
+    #     n: int = 2,
+    #     cluster_field: str = "_cluster_",
+    #     num_clusters: int = 100,
+    #     preprocess_hooks: Optional[List[callable]] = None,
+    # ):
+    #     """
+    #     # TODO:
+    #     """
+    #     pass
+
+    # def _add_cluster_word_cloud_to_config(self, data, cluster_value, top_words):
+    #     # TODO: Add this to wordcloud deployable
+    #     # hacky way I implemented to add top words to config
+    #     data["configuration"]["cluster-labels"][cluster_value] = ", ".join(
+    #         [k for k in top_words if k != "machine learning"]
+    #     )
+    #     data["configuration"]["cluster-descriptions"][cluster_value] = str(top_words)
+
+    # @track
+    # def auto_cluster(
```

Comment:
revert this and ass a deprecated decorator

Reply:
need to discuss further